### ISSUE TITLE
Question about loss2

### ISSUE THREAD

**Issue by Sanchez-web** [benmoseley/harmonic-oscillator-pinn#3]
Hello, your work is very impressive and I learned a lot from it, thank you! 
Here I have one question about loss2. I noticed that the loss2 is mutiply by **1e-4**, I can understand you want to make the training faster, but may I ask why you choose this number and do you have any experience and this figure?
Another question is about the initial conditions, you didn't show them explicit in the code, does it means we do not need to think about these conditions when we use PINN for ODEs?

**Comment by gersonjferreira** [comment]
The initial conditions are **hidden** in the first `loss`. You can reduce the number of points there to

```python
x_data = x[0:2]
y_data = y[0:2]
```
and the code still works.

I wrote (with my student) a version of this code reducing the 2nd order ODE into two 1st order ODEs, then it gets clear where the initial conditions enter. In your code the data is simply the initial position and velocity at t=0.

Regarding **loos2**, indeed the **1e-4** factor is weird and I was also checking that. I use:

```python
    physics = (dx2 + mu*dx + k*yhp) / k
    loss2 = torch.mean(physics**2)
```
Dividing `physics` by k helps because the loss1 and loss2 error are now written in the same units. The **1e-4** term seems to be an implicit workaround to fix the units.

I'll put my code on gitlab soon.

**Comment by Sanchez-web** [comment]
Thanks, your explanation was great!
Looking for

> The initial conditions are **hidden** in the first `loss`. You can reduce the number of points there to
> 
> ```python
> x_data = x[0:2]
> y_data = y[0:2]
> ```
> 
> and the code still works.
> 
> I wrote (with my student) a version of this code reducing the 2nd order ODE into two 1st order ODEs, then it gets clear where the initial conditions enter. In your code the data is simply the initial position and velocity at t=0.
> 
> Regarding **loos2**, indeed the **1e-4** factor is weird and I was also checking that. I use:
> 
> ```python
>     physics = (dx2 + mu*dx + k*yhp) / k
>     loss2 = torch.mean(physics**2)
> ```
> 
> Dividing `physics` by k helps because the loss1 and loss2 error are now written in the same units. The **1e-4** term seems to be an implicit workaround to fix the units.
> 
> I'll put my code on gitlab soon.

**Comment by Sanchez-web** [comment]
Thanks, your explanation was great!
Looking forward to your code!

**Comment by gersonjferreira** [comment]
Thanks. My version is nearly identical to the one here. The only differences are: (i) this normalization of the loss to fix the units/dimensions; (ii) a second version in terms of two 1st order ODEs (reduction of order); (iii) many comments on each step to make it as pedagogical as possible (since this is a project for students entering this research field).

My version is ready, but I want my student to develop her version on her own. So, I'll only upload to gitlab once she finishes her version. It may take a while (couple of weeks maybe), she is still at the beginning of the project.
